Tutorial1-1の課題とコーディング規約の対応

k-shimizu3 opened the pull request:
### 課題
 - Tutorial1-1
 - ホームに「Tutorial1.1」ボタンを配置し AnimationScreenへ遷移する
 - AnimationScreenは[Animations in Flutter](https://flutter.dev/docs/development/ui/animations/tutorial)を参考にする
 - 実装するファイル: lib/animation/animation_screen.dart

---

### 実装内容
main.dart (変更)
 - AnimationScreenに遷移する、Tutorial1.1ボタンの追加
 - ElevatonButtonの作成をヘルパーメソッドからクラスに変更 (コーディング規約対応)

animation/animation_screen.dart (新規作成)
 - 課題で指定されたファイル
 - AnimationScreenの実装

---

### 課題（悩んだこと）
なし

---

### 備考
レビューよろしくお願いします。

k-shimizu3 committed:
Tutorial1-1の課題とコーディング規約の対応

main.dart (変更)
 - AnimationScreenに遷移する、Tutorial1.1ボタンの追加
 - ElevatonButtonの作成をヘルパーメソッドからクラスに変更 (コーディング規約対応)

animation/animation_screen.dart (新規作成)
 - 課題で指定されたファイル
 - AnimationScreenの実装

t-nakata commented on lib/main.dart:
```diff
@@ -34,26 +35,40 @@ class MyHomePage extends StatelessWidget {
           mainAxisAlignment: MainAxisAlignment.center,
           children: <Widget>[
-            _buildElevatedButton(context, 'Tutorial1', BuildingLayoutScreen()),
+            MyElevatedButton(
+              'Tutorial1',
+              BuildingLayoutScreen(),
+            ),
+            const MyElevatedButton(
+              'Tutorial1.1',
+              AnimationScreen(),
+            ),
```
片一方だけにConstつけるのは良くないかな？
両方ともconstつけられるよね？たぶん

k-shimizu3 replied:
おそらくBuildingLayoutScreen()内でStateful Widgetを使っているせいだと思うのですが、一つ目の方にconstをつけるとエラーになってしまったので、つけることができませんでした。

k-shimizu3 replied:
申し訳ありません。
私の勘違いで、constをつけることができました。

`BuildingLayoutScreen`でリンターの指示（`Use key in widget constructors.`）に従うだけでした。
なぜ解決できたのか解らないのでもっと勉強したいと思います。

t-nakata commented on lib/main.dart:
```diff
@@ -34,26 +35,40 @@ class MyHomePage extends StatelessWidget {
-    },
-    child: Text(text),
-  );
+class MyElevatedButton extends StatelessWidget {
+  const MyElevatedButton(this.text, this.newScreen, {Key? key})
```
命名があんまり良くないかもしれないですね。
MyElevatedButtonだとなんの説明にもなっていないのでClassを作るときは、
目的とか用途とかがわかるような名前をつけるようにしてください。

このClassであれば画面遷移を伴うボタンなので、`ScreenNavigatorButton`とかわかりやすいかも？
なにかわかりやすそう名前を付けてください。

k-shimizu3 replied:
レビューありがとうございます！
とても勉強になります！
命名で悩んだのでとりあえず、MyElevatedButtonにしてしまいました。
教えて頂いた`ScreenNavigatorButton `がとても解りやすいので、そちら頂きます！

k-shimizu3 committed:
レビュー修正

main.dart
 - 命名をTransitonScreenButtonに変更

building/building_layout_screen.dart
 - コンストラクタを追加

kenji-axia commented:
@t-nakata 
レビューありがとうございました！
